src/theory/quantifiers/sygus/sygus_grammar_cons.cpp

```diff
@@ -957,7 +957,13 @@ void CegGrammarConstructor::mkSygusDefaultGrammar(
       Trace("sygus-grammar-def") << "...add for singleton" << std::endl;
       std::vector<TypeNode> cargsSingleton;
       cargsSingleton.push_back(unresElemType);
-      sdts[i].addConstructor(SINGLETON, cargsSingleton);
+
+      // lambda x . (singleton (singleton_op T) x) where T = x.getType()
+      Node x = nm->mkBoundVar(etype);
+      Node vars = nm->mkNode(BOUND_VAR_LIST, x);
+      Node singleton = nm->mkSingleton(x.getType(), x);
```

Comment:
```suggestion
      Node singleton = nm->mkSingleton(etype, x);
```

Reply:
Done. In this particular case, x is declared with this type. `etype` is shorter